src/app/container/tool-file-editor/tool-file-editor.component.ts

```diff
@@ -66,6 +66,14 @@ export class ToolFileEditorComponent extends FileEditing {
     return this.currentVersion.id === mostRecentId;
   }
 
+  getIdOfNewestVersion(versions: Array<Tag>): number {
+    return versions.reduce((max, n) => Math.max(max, n.id), this.versions[0].id);
+  }
+
+  getNewestVersion(versions: Array<Tag>): Tag {
```

Comment:
This is the same as the code in [workflow-file-editor.component.ts](https://github.com/dockstore/dockstore-ui2/blob/23d6e519f214dfaec4b9476a0ece63129a96a04a/src/app/workflow/workflow-file-editor/workflow-file-editor.component.ts#L82). It would be good if you could combine the two and put it in a service, then both this file and that file use the common method.